app/javascript/lib/validation/validate-file-content.js

```diff
@@ -511,9 +639,23 @@ export async function parseDenseMatrixFile(chunker, mimeType, fileOptions) {
   return { issues, delimiter, numColumns: headers[0].length }
 }
 
+/** parse a sparse matrix file */
+export async function parseSparseMatrixFile(chunker, mimeType, fileOptions) {
+  let issues = []
+  const delimter = '  '
+  const dataObj = {} // object to track multi-line validation concerns
+  await chunker.iterateLines((rawLine, lineNum, isLastLine) => {
+    issues = issues.concat(validateMTXHeaderLine(rawLine, lineNum, dataObj))
+    issues = issues.concat(validateSparseColumnNumber(rawLine, isLastLine, lineNum, dataObj))
+    issues = issues.concat(validateSparseNoBlankLines(rawLine, isLastLine, lineNum, dataObj))
+    // add other line-by-line validations here
+  })
+  return { issues, delimter, numColumns: dataObj.correctNumberOfColumns}
+}
+
```

Comment:
BLOCKER: This `validate-file-content.js` file is getting too big, let's create a new `expression-matrices-validation.js` file and move all the expression-specific validation functions there

Reply:
Good call I'll make that adjustment

Reply:
For consistency, I'd suggest naming the new file something like `validate-expression-matrices.js` or renaming this file `file-content-validation.js`.

Reply:
I actually wanted to give it a name with a different style, since it's a library of functions used downstream of validate-file-content, not a separate validation method.  But I can see the case for consistency too.